From bd39b27b4f974cd792e810d541e7ad5a1b7d4f61 Mon Sep 17 00:00:00 2001
From: Aleksi Hannula <ahannula4+nixgit@gmail.com>
Date: Tue, 20 Oct 2026 22:50:35 +0300
Subject: [PATCH] minimal-bootstrap.tinycc-{mes,musl*}: fetch from mirror

The upstream repository has Anubis in front of tarball fetches.
---
 pkgs/os-specific/linux/minimal-bootstrap/tinycc/mes.nix  | 8 ++++----
 pkgs/os-specific/linux/minimal-bootstrap/tinycc/musl.nix | 6 +++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/pkgs/os-specific/linux/minimal-bootstrap/tinycc/mes.nix b/pkgs/os-specific/linux/minimal-bootstrap/tinycc/mes.nix
index 6cfe732526714..b7d66ca5a62f2 100644
--- a/pkgs/os-specific/linux/minimal-bootstrap/tinycc/mes.nix
+++ b/pkgs/os-specific/linux/minimal-bootstrap/tinycc/mes.nix
@@ -18,8 +18,8 @@ let
   rev = "cb41cbfe717e4c00d7bb70035cda5ee5f0ff9341";
 
   tarball = fetchurl {
-    url = "https://repo.or.cz/tinycc.git/snapshot/${rev}.tar.gz";
-    hash = "sha256-MRuqq3TKcfIahtUWdhAcYhqDiGPkAjS8UTMsDE+/jGU=";
+    url = "https://github.com/tinycc/tinycc/archive/${rev}.tar.gz";
+    hash = "sha256-c4H5RKqSVc1WDoGSxbAkEkbSyD7qVLjrMXECmS/h4rs=";
   };
   src =
     (kaem.runCommand "tinycc-${version}-source" { } ''
@@ -29,7 +29,7 @@ let
       untar --file ''${NIX_BUILD_TOP}/tinycc.tar
 
       # Patch
-      cd tinycc-${builtins.substring 0 7 rev}
+      cd tinycc-${rev}
       # Static link by default
       replace --file libtcc.c --output libtcc.c --match-on "s->ms_extensions = 1;" --replace-with "s->ms_extensions = 1; s->static_link = 1;"
       replace --file i386-asm.c --output i386-asm.c --match-on "switch(size)" --replace-with "if (reg >= 8) { cstr_printf(add_str, \"%%r%d%c\", reg, (size == 1) ? 'b' : ((size == 2) ? 'w' : ((size == 4) ? 'd' : ' '))); return; } switch(size)"
@@ -37,7 +37,7 @@ let
       # If performing ptr + (-1) for example, the offset should be ptrdiff_t and not size_t
       replace --file tccgen.c --output tccgen.c --match-on "vpush_type_size(pointed_type(&vtop[-1].type), &align);" --replace-with "vpush_type_size(pointed_type(&vtop[-1].type), &align); if (!(vtop[-1].type.t & VT_UNSIGNED)) gen_cast_s(VT_PTRDIFF_T);"
     '')
-    + "/tinycc-${builtins.substring 0 7 rev}";
+    + "/tinycc-${rev}";
 
   meta = {
     description = "Small, fast, and embeddable C compiler and interpreter";
diff --git a/pkgs/os-specific/linux/minimal-bootstrap/tinycc/musl.nix b/pkgs/os-specific/linux/minimal-bootstrap/tinycc/musl.nix
index 4fceaf33471d3..f84b067972b72 100644
--- a/pkgs/os-specific/linux/minimal-bootstrap/tinycc/musl.nix
+++ b/pkgs/os-specific/linux/minimal-bootstrap/tinycc/musl.nix
@@ -15,8 +15,8 @@ let
   rev = "cb41cbfe717e4c00d7bb70035cda5ee5f0ff9341";
 
   src = fetchurl {
-    url = "https://repo.or.cz/tinycc.git/snapshot/${rev}.tar.gz";
-    hash = "sha256-MRuqq3TKcfIahtUWdhAcYhqDiGPkAjS8UTMsDE+/jGU=";
+    url = "https://github.com/tinycc/tinycc/archive/${rev}.tar.gz";
+    hash = "sha256-c4H5RKqSVc1WDoGSxbAkEkbSyD7qVLjrMXECmS/h4rs=";
   };
 
   tccTarget =
@@ -56,7 +56,7 @@ let
       ''
         # Unpack
         tar xzf ${src}
-        cd tinycc-${builtins.substring 0 7 rev}
+        cd tinycc-${rev}
 
         # Patch
         ${lib.concatMapStringsSep "\n" (f: "patch -Np0 -i ${f}") patches}
